test/NuGet.Services.Staging.EndToEndTests/StagingClient/StagingClient.cs

```diff
@@ -0,0 +1,280 @@
+using Newtonsoft.Json.Linq;
+using NuGet.Protocol;
+using Xunit.Abstractions;
+using NuGet.Protocol.Core.v3;
+using System.Linq;
+using System.Net;
+
+namespace NuGet.Services.Staging.EndToEndTests
+{
+    public class StagingClient : IStagingClient
+    {
+        private const string ApiKeyHeader = "X-NuGet-ApiKey";
+        private const string StagePath = "api/stage";
+        private const string PackagePath = "api/package";
+
+        private readonly Uri _stagingServiceUri;
+        private readonly HttpClient _httpClient;
+        private readonly ITestOutputHelper _logger;
+        private readonly NuGetLoggerAdapter _loggerAdapter;
+
+        public StagingClient(Uri stagingServiceUri, ITestOutputHelper logger)
+        {
+            if (stagingServiceUri == null)
+            {
+                throw new ArgumentNullException(nameof(stagingServiceUri));
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            _stagingServiceUri = stagingServiceUri;
+            _logger = logger;
+            _loggerAdapter = new NuGetLoggerAdapter(_logger);
+
+            _httpClient = new HttpClient();
+        }
+
+        public async Task<JArray> ListUserStages(string apiKey)
+        {
+            _logger.WriteLine("StagingClient: ListUserStages called.");
+
+            Func<HttpRequestMessage> requestFactory = () =>
+            {
+                
+                var request = new HttpRequestMessage(HttpMethod.Get, new Uri(_stagingServiceUri, StagePath));
+                request.Headers.Add(ApiKeyHeader, apiKey);
+
+                return request;
+            };
+
+            var responseBody = await SendAsync(requestFactory);
+
+            return JArray.Parse(responseBody);
+        }
+
+        public async Task<JObject> GetDetails(string stageId)
+        {
+            _logger.WriteLine($"StagingClient: GetDetails called for stage {stageId}");
+
+            Func<HttpRequestMessage> requestFactory = () =>
+            {
+                var request = new HttpRequestMessage(HttpMethod.Get, new Uri(_stagingServiceUri, $"{StagePath}/{stageId}"));
+                return request;
+            };
+
+            var responseBody = await SendAsync(requestFactory);
+
+            return JObject.Parse(responseBody);
+        }
+
+        public async Task<JObject> CreateStage(string displayName, string apiKey)
+        {
+            _logger.WriteLine($"StagingClient: CreateStage called with name {displayName}");
+
+            Func<HttpRequestMessage> requestFactory = () =>
+            {
+                var request = new HttpRequestMessage(HttpMethod.Post, new Uri(_stagingServiceUri, StagePath));
+                request.Headers.Add(ApiKeyHeader, apiKey);
+
+                request.Content = new StringContent("\"" + displayName + "\"", Encoding.UTF8);
+                request.Content.Headers.ContentType.MediaType = "application/json";

```

Comment:
fixed

---
In reply to: [64174010](https://github.com/NuGet/NuGet.Staging/pull/49#discussion_r64174010) [](ancestors = 64174010)